### DISCUSSION TITLE
OyasumiVR: VR Sleeping Utilities

### DISCUSSION THREAD

**Discussion by Raphiiko** [vrchat-community/osc#156] [Show and tell]
<h1>
  <img src="https://github.com/Raphiiko/OyasumiVR/assets/111654848/53f56237-9cde-48df-a446-8eaad76c8206">
</h1>

<h3 align="center">
    :zzz: A utility to assist with sleeping in virtual reality. :zzz:
</h3>

## What is OyasumiVR?

OyasumiVR is a desktop application containing various utilities and automations to **help you sleep with a virtual reality headset on**. Most of these utilities are built specifically for use within the social vr-platform [VRChat](https://vrchat.com). 

OyasumiVR can **detect when you are asleep**, and help you in various ways such as:
- :battery: Saving electricity and battery overnight
- :low_brightness: Dimming your headset brightness
- :sparkles: Triggering avatar animations based on your sleeping pose
- :envelope_with_arrow: Automatically handling invite requests
- :hammer_and_wrench: Automating various SteamVR settings
  
...and many more! 

<h2 align="center">Features specifically related to VRChat OSC<br>
<h3 align="center"><a href="https://github.com/Raphiiko/OyasumiVR/tree/develop#readme">For the full feature list, please check our readme!</a></h3>

### 💫 VRChat Automations
<table>
  <tr>
    <td>
      🌜 <b>Sleeping Animations</b>
      <br>
      OyasumiVR can detect what sleeping pose you are in, and use it to trigger the right animation or pose on your avatar. 
      <br><br>
      There are built-in presets for popular locomotion assets, including:
      <br><br>
      <ul>
        <li>
          <a href="https://minminmart.booth.pm/items/2886739">ごろ寝システム v2.2 - 2.3</a> by <a href="https://twitter.com/minminmeeean">みんみんみーん</a>
          <br>(Sleep System v2.2 - 2.3 by minminmiin)
          <br><a href="https://booth.pm/en/items/4233545">EX version</a> also supported!
        </li>
        <li>
          <a href="https://booth.pm/en/items/3290806">GoGo Loco v1.7.1+</a> by <a href="https://twitter.com/franadaVRC">franada</a>
        </li>
      </ul>
      <br>
      Alternatively, trigger your own animations with the included OSC script editors!
    </td>
    <td width="190">
      <img src="https://user-images.githubusercontent.com/111654848/198073951-85a52109-6fad-4edd-ad53-897a6d8adf12.gif">
    </td>
    <td width="190">
      <img src="https://github.com/Raphiiko/OyasumiVR/assets/111654848/1004dbea-2096-4b29-9791-eb78403a63a5">
    </td>
  </tr>
  <tr>
    <td>
      🔧 <b>OSC Automations</b>
      <br>
      You can send certain OSC messages to VRChat when sleep mode is enabled, or disabled. You can use the built-in OSC script editor to compose the exact messages you want to send.
      <br><br>
      This is a more advanced feature that you can use use to trigger certain features on your avatar. For example: Show a sleepy expression on your face while sleep mode is enabled!
    </td>
    <td width="380" colspan="2"><img src="https://github.com/Raphiiko/OyasumiVR/assets/111654848/dd98c34b-2cce-4037-aaa9-85cdf722cd83"></td>
  </tr>
</table>

### OSC Script Editing
For these features, OyasumiVR contains a OSC script editor to facilitate integration with VRChat's OSC functionality. Here's some screenshots of that! More information about this can be found in the [OSC Script Editor & Syntax Guide](https://github.com/Raphiiko/OyasumiVR/wiki/OSC-Script-Editor-&-Syntax-Guide)

<table>
<tr><th>Simple Editor</th><th>Advanced Editor</th></tr>
<tr>
<td><img src="https://github.com/vrchat-community/osc/assets/111654848/27f47003-12cb-4bca-91db-9f5201125075"></td>
<td><img src="https://github.com/vrchat-community/osc/assets/111654848/6bbb6d0d-9f78-4977-9864-ec353862518e"></td>
</tr>
</table>

**Comment by Sharrnah** [comment]
@Raphiiko I love this. Make sure to vote for https://feedback.vrchat.com/website-bug-reports/p/osc-resources-list-is-never-updated
as it seems they currently never update the listing.
